staging/src/k8s.io/cloud-provider/fake/fake.go

```diff
@@ -321,13 +323,20 @@ func (f *Cloud) InstanceMetadata(ctx context.Context, node *v1.Node) (*cloudprov
 	f.addCall("instance-metadata-by-provider-id")
 	f.addressesMux.Lock()
 	defer f.addressesMux.Unlock()
+
+	providerID := node.Spec.ProviderID
```

Comment:
I don't think the fake cloud provider should inherit what's in `node.Spec.ProviderID`. That is actually logic in the cloud node controller that we want to test against.

Reply:
ahh gotcha, updated!